padertorch/data/fragmenter.py

```diff
@@ -109,86 +119,145 @@ def fragment(key, x):
 
-    if offset_mode == 'start':
+    if mode == 'begin':
         start = 0
-    elif offset_mode == 'random':
-        start = np.random.randint(num_samples - length)
-    elif offset_mode in ['end', 'center']:
+    elif mode == 'random':
+        start = rng.randint(shift)
+    elif mode == 'random_max_segments':
+        start = rng.randint((num_samples - length) % shift)
+    elif mode in ['end', 'center']:
         remainder = (num_samples - length) % shift
-        if offset_mode == 'end':
+        if mode == 'end':
             start = remainder
         else:
             start = remainder // 2
     else:
-        raise ValueError('Unknown offset mode', offset_mode)
-    offset = np.arange(start, num_samples - length + 1, shift)
-
-    if num_segments:
-        max_segments = (num_samples - length) // shift + 1
-        assert num_segments <= max_segments, (num_segments, max_segments)
-        offset = offset[:num_segments]
-    return offset
+        raise ValueError('Unknown offset mode', mode,
+                         'choose on of', possible_boundary_modes)
+    start = np.arange(start, num_samples - length + 1, shift)
+    if choose_one:
+        start = rng.choice(start, 1)
+    stop = start + length
+    boundaries = np.stack([start, stop], axis=0)
+    return boundaries
 
 
-def segment(x, length, offset=None, shift=None, axis=-1,
-            offset_mode='start', num_segments=None):
+def segment(x, axis=-1, boundary=None, length: int=None, shift=None,
+            mode='begin', choose_one=False, rng=np.random):
     """
+    Segments a signal x along an axis. Either with predefined segment
+    boundaries if boundary is set or with internally calculated boundaries if
+    length is set.
 
     Args:
-        x:
-        segment_length:
-        step:
-        axis:
-        offset_mode:
-        num_segments:
+        x: signal to be segmented
+        axis: axis which is segmented
+        boundary: 2xB numpy array with start and end values
+            for B pre calculated boundaries, which may be used to segment
+            multiple audios in the same way.
+        length: segment length
+        shift: shift between segments
+        mode: Defines the position of the boundaries in the signal:
+            begin: boundaries start at sample zero
+                so that only values at the end are cut
+            end: boundaries end at num_samples
+                so that only values at the beginning are cut
+            center: boundaries are chosen such that the same number of samples
+                are discarded at the end and the beginning of the signal
+            random: Starts the first segment randomly at a value
+                between 0 and shift
+            random_max_segments: Randomly chooses the first segment such that
+                the maximum number of segments are created
+        choose_one: If True only outputs one segment boundary
+        rng: random number generator (numpy.random)
```

Comment:
Backticks for `numpy.random`, `shift`, ... all arguemnts references in the text

Reply:
done